nncf/torch/pruning/utils.py

```diff
@@ -61,3 +67,76 @@ def init_output_masks_in_graph(graph: NNCFGraph, nodes: List):
+
+    :param graph: the model graph
+    :param node: node from NNCF graph
+    :return: output shape
+    """
+    in_edge = graph.get_input_edges(node)[0]
+    shape = list(in_edge.tensor_shape)[2:]
+    attrs = node.layer_attributes
+
+    assert isinstance(attrs, ConvolutionLayerAttributes)
+
+    for i, _ in enumerate(shape):
+        if attrs.transpose:
+            shape[i] = (shape[i] - 1) * attrs.stride[i] - 2 * attrs.padding_values[i] + attrs.kernel_size[i]
+        else:
+            shape[i] = (shape[i] + 2 * attrs.padding_values[i] - attrs.kernel_size[i]) // attrs.stride[i] + 1
+    return tuple(shape)
+
+
+def collect_output_shapes(graph: NNCFGraph) -> Dict[NNCFNodeName, List[int]]:
+    """
+    Collects output dimension shapes for convolutions and fully connected layers
+    from the connected edges in the NNCFGraph.
+
+    :param graph: NNCFGraph.
+    :return: Dictionary of output dimension shapes. E.g {node_name: (height, width)}
+    """
+    modules_out_shapes = {}
+    for node in graph.get_nodes_by_types([v.op_func_name for v in NNCF_GENERAL_CONV_MODULES_DICT]):
+        output_edges = graph.get_output_edges(node)
+        if output_edges:
+            out_edge = output_edges[0]
+            out_shape = out_edge.tensor_shape[2:]
+        else:
+            # For disconnected NNCFGraph when node have no output edge
+            out_shape = _calculate_output_shape(graph, node)
+            nncf_logger.error("Node %s have no output edge in NNCFGraph", node.node_name)
+        modules_out_shapes[node.node_name] = out_shape
+
+    for node in graph.get_nodes_by_types([v.op_func_name for v in NNCF_LINEAR_MODULES_DICT]):
+        output_edges = graph.get_output_edges(node)
+        if output_edges:
+            out_edge = graph.get_output_edges(node)[0]
+            out_shape = out_edge.tensor_shape
+            modules_out_shapes[node.node_name] = out_shape[-1]
+        else:
+            # For disconnected NNCFGraph when node have no output edge
+            nncf_logger.error("Node %s have no output edge in NNCFGraph", node.node_name)
+            modules_out_shapes[node.node_name] = node.layer_attributes.out_features
+    return modules_out_shapes
+
+
+def collect_input_shapes(graph: NNCFGraph) -> Dict[NNCFNodeName, List[int]]:
+    """
+    Collects input dimension shapes for fully connected layers from the connected edges in the NNCFGraph.
+
+    :param graph: NNCFGraph.
+    :return: Dictionary of input dimension shapes. E.g {node_name: (height, width)}
+    """
+    modules_in_shapes = {}
+    for node in graph.get_nodes_by_types([v.op_func_name for v in NNCF_LINEAR_MODULES_DICT]):
+        in_edge = graph.get_input_edges(node)[0]
+        in_shape = in_edge.tensor_shape
+        if len(in_shape) == 1:
+            modules_in_shapes[node.node_name] = in_shape[0]
+        else:
+            modules_in_shapes[node.node_name] = in_shape[1:]
+    return modules_in_shapes
```

Comment:
Any file called `utils.py` should not be extended. Introduce a new entity encapsulating the new functionality, separate the code into an own .py file and write corresponding unit tests.

Reply:
It was part of the filter pruning algorithm. I just moved it to common. I hope it's already covered by unit tests. @daniil-lyakhov could you confirm that? 
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/pruning/filter_pruning/algo.py#L286

Reply:
I agree that utils.py is more appropriate place for this functions. We have test which checks mask propagation in disconnected graphs, but tests of flops counting for disconnected graphs are missing. I've reflected it in my ticket 70128

Reply:
Thanks, @daniil-lyakhov